internal/fdo_test.go

```diff
@@ -0,0 +1,204 @@
+}
+
+func setTestEnv() {
+	workingDir, err := os.Getwd()
+	if err != nil {
+		fyne.LogError("Could not get current working directory", err)
+	}
+	err = os.Setenv("XDG_DATA_DIRS", filepath.Join(workingDir, "testdata"))
+	if err != nil {
+		fyne.LogError("Could not set test environment variable", err)
+	}
+}
+
+//applications/app1.desktop and icons/default_theme/apps/32x32/app1.png
+func TestFdoLookupDefaultTheme(t *testing.T) {
+	setTestEnv()
+	data := fdoLookupApplication(iconTheme, iconSize, "app1")
+	assert.Equal(t, exists(data), true)
+}
+
+//applications/com.fyne.app.desktop and icons/default_theme/apps/scalable/app2.svg
+func TestFdoFileNameMisMatchAndScalable(t *testing.T) {
+	setTestEnv()
+	data := fdoLookupApplication(iconTheme, iconSize, "app2")
+	assert.Equal(t, exists(data), true)
+}
+
+//applications/app3.desktop and applications/app3.png
+func TestFdoIconNameIsPath(t *testing.T) {
+	setTestEnv()
+	dataLocation := os.Getenv("XDG_DATA_DIRS")
+	output := fmt.Sprintf("[Desktop Entry]\nName=App3\nExec=app3\nIcon=%s\n", filepath.Join(dataLocation, "icons", "app3.png"))
+	fmt.Print(output)
+	err := ioutil.WriteFile(filepath.Join(dataLocation, "applications", "app3.desktop"), []byte(output), 0644)
+	if err != nil {
+		fyne.LogError("Could not create desktop for Icon Name path example", err)
+	}
+	data := fdoLookupApplication(iconTheme, iconSize, "app3")
+	fmt.Println(data.IconPath())
+	assert.Equal(t, exists(data), true)
+}
+
+//applications/app4.desktop and pixmaps/app4.png
+func TestFdoIconInPixmaps(t *testing.T) {
+	setTestEnv()
+	data := fdoLookupApplication(iconTheme, iconSize, "app4")
+	assert.Equal(t, exists(data), true)
+}
+
+//applications/app5.desktop and icons/hicolor/32x32/apps/app5.png
+func TestFdoIconHicolorFallback(t *testing.T) {
+	setTestEnv()
+	data := fdoLookupApplication(iconTheme, iconSize, "app5")
+	assert.Equal(t, exists(data), true)
+}
+
+//applications/app6.desktop and icons/hicolor/scalable/apps/app6.svg
+func TestFdoIconHicolorFallbackScalable(t *testing.T) {
+	setTestEnv()
+	data := fdoLookupApplication(iconTheme, iconSize, "app6")
+	assert.Equal(t, exists(data), true)
+}
+
+//applications/app7.desktop and icons/default_theme/apps/16x16/app7.png
+func TestFdoLookupDefaultThemeDifferentSize(t *testing.T) {
+	setTestEnv()
+	data := fdoLookupApplication(iconTheme, iconSize, "app7")
+	assert.Equal(t, exists(data), true)
+}
+
+//applications/app8.desktop and icons/third_theme/apps/32/app8.png
+func TestFdoLookupAnyThemeFallback(t *testing.T) {
+	setTestEnv()
+	data := fdoLookupApplication(iconTheme, iconSize, "app8")
+	assert.Equal(t, exists(data), true)
+}
+
+//applications/app9.desktop and icons/third_theme/emblems/16x16/app9.png
+func TestFdoLookupIconNotInApps(t *testing.T) {
+	setTestEnv()
+	data := fdoLookupApplication(iconTheme, iconSize, "app9")
+	assert.Equal(t, exists(data), true)
+}
+
+func TestFdoLookupIconByWinInfo(t *testing.T) {
+	setTestEnv()
+	//Test win info lookup by title
+	win1 := &dummyWindow1{}
```

Comment:
As above the multiple structs make it hard to read this - "dummyWindow1" could be "dummyWIndow{title="blah"}" - then we have test setup next to test code